tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,314 @@
+from time import sleep
+
+from krbd_io_handler import krbd_io_handler
+
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def device_cleanup(rbd, file_name, **kw):
+    """
+    Unmout the given file_name, unmap the given device_name encrypted using encryption_config
+    Args:
+        rbd: rbd object
+        file_name: <path>/<file> to be unmounted
+        kw:
+            "image_spec": image which was encrypted
+            "encryption_config": {<type>:<passphrase>} used for encryption
+            "device_type": default "nbd"
+            "device_name": device created for image map without encryption
+    """
+    flag = 0
+    if rbd.exec_cmd(
+        cmd=f"umount {file_name}",
+        sudo=True,
+    ):
+        log.error(f"Umount failed for {file_name}")
+        flag = 1
+
+    if kw.get("image_spec"):
+        if rbd.device_unmap(
+            f"{kw['image_spec']}", kw.get("device_type", "nbd"), kw["encryption_config"]
+        ):
+            log.error(
+                f"Device unmap failed for {kw['image_spec']} with encryption config {kw['encryption_config']}"
+            )
+            flag = 1
+
+    return flag
+
+
+def test_rbd_encryption(rbd, pool_type, format, **kw):
+    """
+    Perform luks encryption on the given rbd pool, image and its clone and verify data integrity
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        format: encryption format of parent and clone Ex: luks1,luks2
+        **kw: Test data
+    """
+    [parent_encryption_type, clone_encryption_type] = format.split(",")
+    pool = kw["config"][pool_type]["pool"]
+    image = f"{parent_encryption_type}_{rbd.random_string()}"
+    size = "1G"
+    kw["config"][pool_type]["size"] = size
+    snap = f"{clone_encryption_type}_snap_{rbd.random_string()}"
+    clone = f"{clone_encryption_type}_clone_{rbd.random_string()}"
+    image_spec = f"{pool}/{image}"
+    clone_spec = f"{pool}/{clone}"
+    snap_spec = f"{pool}/{image}@{snap}"
+    encryption_config = list()
+    parent_file_path = f"/tmp/{image}_dir/{image}_file"
+    clone_file_path = f"/tmp/{clone}_dir/{image}_file"
+
+    try:
+        io_config = {
```

Comment:
it's good to pass `io_config` from test suite so in future according to the cluster size we can scale up.

Reply:
As of now IO size is fixed at 100M, we can modify this when we are automating tier-3 scenarios for encryption which do not exist even in polarion as of not. 
Can be taken up as part of different PR IMO